float32 diff_x
float32 diff_y
float32 diff_z
float32 diff_yaw
